akka-docs/src/test/java/jdocs/persistence/testkit/TestKitExamples.java

```diff
@@ -0,0 +1,193 @@
+      // check the type of operation and react with success or with reject or with failure.
+      // if you return ProcessingSuccess the operation will be performed, otherwise not.
+      if (count < 10) {
+        count += 1;
+        if (processingUnit instanceof ReadEvents) {
+          ReadEvents read = (ReadEvents) processingUnit;
+          if (read.batch().nonEmpty()) {
+            ProcessingSuccess.getInstance();
+          } else {
+            return StorageFailure.create();
+          }
+        } else if (processingUnit instanceof WriteEvents) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit instanceof DeleteEvents) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit.equals(ReadSeqNum.getInstance())) {
+          return Reject.create();
+        }
+        // you can set your own exception
+        return StorageFailure.create(new RuntimeException("your exception"));
+      } else {
+        return ProcessingSuccess.getInstance();
+      }
+    }
+  }
+  // #set-event-storage-policy
+
+  // #set-snapshot-storage-policy
+  class SnapshotStoragePolicy implements ProcessingPolicy<SnapshotOperation> {
+
+    // you can use internal state, it doesn't need to be thread safe
+    int count = 1;
+
+    @Override
+    public ProcessingResult tryProcess(String persistenceId, SnapshotOperation processingUnit) {
+      // check the type of operation and react with success or with failure.
+      // if you return ProcessingSuccess the operation will be performed, otherwise not.
+      if (count < 10) {
+        count += 1;
+        if (processingUnit instanceof ReadSnapshot) {
+          ReadSnapshot read = (ReadSnapshot) processingUnit;
+          if (read.getSnapshot().isPresent()) {
+            ProcessingSuccess.getInstance();
+          } else {
+            return StorageFailure.create();
+          }
+        } else if (processingUnit instanceof WriteSnapshot) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit instanceof DeleteSnapshotsByCriteria) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit instanceof DeleteSnapshotByMeta) {
+          return ProcessingSuccess.getInstance();
+        }
+        // you can set your own exception
+        return StorageFailure.create(new RuntimeException("your exception"));
+      } else {
+        return ProcessingSuccess.getInstance();
+      }
+    }
+  }
+  // #set-snapshot-storage-policy
+
+}
+
+class Cmd {
+
+  private String data;
+
+  Cmd(String data) {
+    this.data = data;
+  }
+
+  public String getData() {
+    return data;
+  }
+
+  public void setData(String data) {
+    this.data = data;
+  }
+}
+
+class Evt {
+
+  private String data;
+
+  Evt(String data) {
+    this.data = data;
+  }
+
+  public String getData() {
+    return data;
+  }
+
+  public void setData(String data) {
+    this.data = data;
+  }
+}
+
+class EmptyState {}
```

Comment:
rename to `State`

also move State, Evt and Cmd into YourPersistentBehavior as `static class`